### PR TITLE
Including ToC in Best Practices

### PR DESCRIPTION
**Pull request by mmattel**
This is a short description for the use of `Table of Contents` in Best Practices (documentation in the repro, not adoc!)

### CODE REVIEW

File: docs/best-practices.md [hunk 1]
```diff
@@ -155,6 +158,31 @@ Example:
 `include::encryption-types.adoc[leveloffset=+1]`
 (the including file in this example is in the same directory as the included file)
 
+## Table of Contents
+
+Prefix: `toc`
+
+Refernce: [`Table of Contents (ToC)`](https://asciidoctor.org/docs/user-manual/#user-toc)
```
Comment (settermjd) [inline docs/best-practices.md]: s/Refernce/Reference/g

File: docs/best-practices.md [hunk 2]
```diff
@@ -88,6 +89,8 @@ You can reference a section or an anchor inside the same file, another file - ev
 
 Example: `xref:configuration/server/occ_command.adoc#apps-commands[the Market app]`
 
+**Advise** Please consider using a ToC ([Table of Contents](#table-of-contents)) instead a list of xref´s.
```
Comment (settermjd) [inline docs/best-practices.md]: Strongly in favour of this, where relevant.

File: docs/best-practices.md [hunk 3]
```diff
@@ -155,6 +158,31 @@ Example:
...
+
+Prefix: `toc`
+
+Refernce: [`Table of Contents (ToC)`](https://asciidoctor.org/docs/user-manual/#user-toc)
+
+A table of contents (ToC) is, if not otherwise defined, an index of section and subsection
+titles that can be automatically generated from the pages structure when converting a
+document with Asciidoctor.
+
+The easiest way adding a ToC is shown in the following example.
```
Comment (settermjd) [inline docs/best-practices.md]: It should start as "The easiest way of adding a ToC".

File: docs/best-practices.md [hunk 4]
```diff
@@ -155,6 +158,31 @@ Example:
...
+```
+= The Page Header
+:toc:
+
+Page contents.
+```
+Please also see additional directives like:
+[`toc-title`](https://asciidoctor.org/docs/user-manual/#user-toc-title),
+[`toc-level`](https://asciidoctor.org/docs/user-manual/#user-toc-levels) or
+[`In-Document Placement`](https://asciidoctor.org/docs/user-manual/#manual-placement)
```
Comment (settermjd) [inline docs/best-practices.md]: Why is the text for this link wrapped in backticks?

Comment (mmattel) [reply docs/best-practices.md]: Because we did on top of the page for all the other references too and the page renders in github correct

Comment (settermjd) [reply docs/best-practices.md]: But the text is normal wording, not a technical term, nor reference.

Comment (mmattel) [reply docs/best-practices.md]: I have tried without square brackets respectively without the ticks, but then it does not render correct. Leaving it as committed.

**Comment by mmattel** [comment]
@settermjd, fixed all issues, ready for review

**Pull request merged**